fs/xfstests.py

```diff
@@ -39,14 +39,27 @@ def setUp(self):
         Source: git://git.kernel.org/pub/scm/fs/xfs/xfstests-dev.git
         """
         sm = SoftwareManager()
-        packages = ['xfslibs-dev', 'uuid-dev', 'libtool-bin', 'e2fsprogs',
-                    'automake', 'gcc', 'libuuid1', 'quota', 'attr',
-                    'libattr1-dev', 'make', 'libacl1-dev', 'xfsprogs',
-                    'libgdbm-dev', 'gawk', 'fio', 'dbench', 'uuid-runtime']
+        detected_distro = distro.detect()
+
+        packages = ['e2fsprogs', 'automake', 'gcc', 'quota', 'attr',
+                    'make',  'xfsprogs',  'gawk', 'fio', 'dbench']
+
+        if 'Ubuntu' in detected_distro.name:
+            packages.extend(['xfslibs-dev', 'uuid-dev', 'libtool-bin', 'libuuid1',
+                             'libattr1-dev', 'libacl1-dev', 'libgdbm-dev', 'uuid-runtime'])
+
+        elif detected_distro.name in ['centos', 'fedora', 'redhat']:
+            packages.extend(['acl', 'bc', 'dump', 'indent', 'libtool', 'lvm2',
+                             'xfsdump', 'psmisc', 'sed', 'libacl-devel', 'libattr-devel',
+                             'libaio-devel', 'libuuid-devel', 'openssl-devel', 'xfsprogs-devel',
+                             'btrfs-progs-devel'])
+        else:
+            self.skip("test not supported in %s" % detected_distro.name)
+
         for package in packages:
             if not sm.check_installed(package) and not sm.install(package):
-                self.error("Fail to install %s required for this test." %
-                           package)
+                self.skip("Fail to install %s required for this test." %
```

Comment:
shouldn't it be error, as we skip tests if package is not there or if env is unsupported, but when test tries to install some package and if it fails to install its better to error out.